nussl/separation/duet.py

```diff
@@ -0,0 +1,440 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import axes3d
+from scipy import signal
+
+import nussl.audio_signal
+import nussl.constants
+import nussl.spectral_utils
+import nussl.utils
+import mask_separation_base
+import masks
+
+
+class Duet(mask_separation_base.MaskSeparationBase):
+
+    """Implements the Degenerate Unmixing Estimation Technique (DUET) algorithm.
+
+    The DUET algorithm was originally proposed by S.Rickard and F.Dietrich for DOA estimation
+    and further developed for BSS and demixing by A.Jourjine, S.Rickard, and O. Yilmaz.
+
+    References:
+
+        * Rickard, Scott. "The DUET blind source separation algorithm." Blind Speech Separation. Springer Netherlands,
+          2007. 217-241.
+        * Yilmaz, Ozgur, and Scott Rickard. "Blind separation of speech mixtures via time-frequency masking." Signal
+          Processing, IEEE transactions on 52.7 (2004): 1830-1847.
+
+    Parameters:
+        input_audio_signal (np.array): a 2-row Numpy matrix containing samples of the two-channel mixture.
+        num_sources (int): Number of sources to find.
+        attenuation_min (int): Lower bound on attenuation value, used as minimum distance in utils.find_peak_indices.
+        attenuation_max (int): Upper bound on attenuation, used for creating a histogram without outliers.
+        num_attenuation_bins (int): Number of bins for attenuation. 
+        delay_min (int): Lower bound on delay, used as minimum distance in utils.find_peak_indices.
+        delay_max (int): Upper bound on delay, used for creating a histogram without outliers.
+        num_delay_bins (int): Number of bins for delay. 
+        peak_threshold (float): Value in [0, 1] for peak picking. 
+        attenuation_min_distance (int): Minimum distance between peaks wrt attenuation. 
+        delay_min_distance (int): Minimum distance between peaks wrt delay.
+        p (int): Weight the histogram with the symmetric attenuation estimator
```

Comment:
I remember the paper had some recommendations for values for these two variables. Could you reproduce those recommendations here? (Make sure to cite the paper if you decide to directly copy/paste, else paraphrasing is fine.)